### PR TITLE
Ported survival02 to lua

### PR DESCRIPTION
**Pull request by abcdefg30**
What this PR does:
- Ported survival02 to lua
- Added a lua functions for airstrikes (spyplane, parabombs)
- Added a lua function for paratroopers and replaced the "old paradrop methods"

Still to do:
- [x] Teach the lua-c# bridge how to use nullable types (WPos?) - _this PR does not needed that anymore_
- [x] [This](https://github.com/abcdefg30/OpenRA/commit/be6cb0537635f5aae7f11caf93db310213d45572#diff-61cd027831d09aa05e113bf2de1593a4R149) doesn't seem to be done correctly, I'll have another look at it -- _should be fixed now_
- [x] Soviets are _no longer_ killed by their own mines
- [x] Add_ed_ guards for the soviet harvesters

### CODE REVIEW

File: mods/cnc/maps/gdi04b/map.yaml [hunk 1]
```diff
@@ -583,30 +541,24 @@ Actors:
...
-	waypoint3: waypoint
+	NodPatrol2: waypoint
 		Location: 11,42
 		Owner: Neutral
-	waypoint2: waypoint
+	NodPatrol3: waypoint
 		Location: 12,28
 		Owner: Neutral
-	waypoint1: waypoint
+	NodPatrol4: waypoint
```
Comment (abcdefg30) [inline mods/cnc/maps/gdi04b/map.yaml]: Thanks. Fixed in #7261.

File: mods/ra/maps/survival02/map.yaml [hunk 2]
```diff
@@ -0,0 +1,1140 @@
...
+			Prerequisites: ~disabled
+	TTNK:
+		Buildable:
+			Prerequisites: ~disabled
+	CTNK:
+		Buildable:
+			Prerequisites: ~disabled
+	DOME:
+		Buildable:
+			Prerequisites: ~disabled
```
Comment (abcdefg30) [inline mods/ra/maps/survival02/map.yaml]: I'm not sure as I didn't do the rules. But removed it.

**Comment by abcdefg30** [comment]
Finally, updated.
Fixed all nits, splitted the PR in this one, #7261 and #7269 .
Added guards for the harvesters and
updated the year numbers in the directories I touched files.

**Comment by Mailaender** [comment]
Crash when I tried to repair the southern middle bridge:

```
Fatal Lua Error: [string "survival02.lua"]:225: attempt to index global 'Brigde2ReinforcementsEntry' (a nil value)
   at OpenRA.Scripting.ScriptContext.FatalError(System.String message)
   at OpenRA.Mods.Common.Scripting.ScriptTriggers.Damaged(OpenRA.Actor self, OpenRA.Traits.AttackInfo e)
   at OpenRA.Traits.Health.Resurrect(OpenRA.Actor self, OpenRA.Actor repairer)
   at OpenRA.Mods.RA.Traits.Bridge+<Repair>c__AnonStorey2.<>m__0(OpenRA.World w)
   at OpenRA.World.Tick()
   at OpenRA.Game.InnerLogicTick(OpenRA.Network.OrderManager orderManager)
   at OpenRA.Game.LogicTick()
   at OpenRA.Game.Loop()
   at OpenRA.Game.Run()
   at OpenRA.Program.Run(System.String[] args)
   at OpenRA.Program.Main(System.String[] args)
```

**Comment by abcdefg30** [comment]
Sorry, updated. (I just renamed the waypoints, seems like the names have been too long.)

**Comment by Mailaender** [comment]
Needs a rebase now.

**Comment by abcdefg30** [comment]
Rebased.

**Comment by Mailaender** [comment]
The notifications you set up and deleted again https://github.com/OpenRA/OpenRA/pull/7164#issuecomment-68109757 are actually fine. It would be great if you could come back with a new pull request for them. Even though we don't use them here, others mappers might want them and it is also far more convenient than looking them up with the Windows only XCC mixer through wine.

**Comment by abcdefg30** [comment]
Rebased. I filed a new PR for the notifications: #7287

**Comment by Mailaender** [comment]
I can confirm that the bridge repair objective works again and all touched missions have working paradrops. Did not test the Intervention mission.

**Comment by Mailaender** [comment]
This needs a rebase now.

**Comment by abcdefg30** [comment]
Rebased.
Added tooltips to all fence types that haven't had one before as discussed in IRC.

**Comment by obrakmann** [comment]
:+1: looks good to me. Tested the other affected missions, tested air strikes and paratroopers in skirmish and missions.

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/6237d3db2ccbc47a29fbee78eadda99153e631e5...32aec171ad27f16a0b5f056b58984d8249351471)

**Pull request merged**